aws-cpp-sdk-transfer/source/transfer/TransferManager.cpp

```diff
@@ -268,7 +268,12 @@ namespace Aws
                     PartPointer partPtr = partsIter->second;
                     Aws::S3::Model::UploadPartRequest uploadPartRequest = m_transferConfig.uploadPartTemplate;
                     uploadPartRequest.SetContinueRequestHandler([handle](const Aws::Http::HttpRequest*) { return handle->ShouldContinue(); });
-                    uploadPartRequest.SetDataSentEventHandler([handle, partPtr](const Aws::Http::HttpRequest*, long long amount){ partPtr->OnDataTransferred(amount, handle); });
+                    auto dataEventHandler = [this, handle, partPtr](const Aws::Http::HttpRequest*, long long amount)
+                    {
```

Comment:
My first attempt did exactly what you suggest and I "improved" it in the final version of what I submitted in the PR! 

I'll happily redo it when I get home later tonight, thanks!